x-pack/legacy/plugins/ml/public/application/datavisualizer/index_based/components/search_panel/search_panel.tsx

```diff
@@ -52,94 +70,81 @@ export const SearchPanel: FC<Props> = ({
-              defaultMessage: 'Currently only KQL saved searches can be edited',
-            })}
-          >
-            <EuiFieldSearch
-              value={`${searchString}`}
-              readOnly
-              data-test-subj="mlDataVisualizerLuceneSearchBarl"
-            />
-          </EuiFormRow>
-        </EuiForm>
-      )}
-      <EuiSpacer size="l" />
-      <EuiFlexGroup gutterSize="xs" alignItems="center">
-        <EuiFlexItem grow={false}>
-          <FormattedMessage
-            id="xpack.ml.datavisualizer.searchPanel.sampleLabel"
-            defaultMessage="Sample"
-          />
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiSelect
-            options={searchSizeOptions}
-            value={samplerShardSize}
-            onChange={e => setSamplerShardSize(+e.target.value)}
-            aria-label={i18n.translate('xpack.ml.datavisualizer.searchPanel.sampleSizeAriaLabel', {
-              defaultMessage: 'Select number of documents to sample',
-            })}
-            data-test-subj="mlDataVisualizerShardSizeSelect"
+    <EuiFlexGroup gutterSize="m" alignItems="center" data-test-subj="mlDataVisualizerSearchPanel">
+      <EuiFlexItem>
+        {searchQueryLanguage === SEARCH_QUERY_LANGUAGE.KUERY ? (
+          <KqlFilterBar
+            indexPattern={indexPattern}
+            onSubmit={searchHandler}
+            initialValue={searchString}
+            placeholder={i18n.translate(
+              'xpack.ml.datavisualizer.searchPanel.queryBarPlaceholderText',
+              {
+                defaultMessage: 'Search… (e.g. status:200 AND extension:"PHP")',
+              }
+            )}
           />
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <span>
-            <FormattedMessage
-              id="xpack.ml.datavisualizer.searchPanel.documentsPerShardLabel"
-              defaultMessage="documents per shard from a total of {wrappedTotalCount} {totalCount, plural, one {document} other {documents}}"
-              values={{
-                wrappedTotalCount: (
-                  <b data-test-subj="mlDataVisualizerTotalDocCount">{totalCount}</b>
-                ),
-                totalCount,
-              }}
+        ) : (
+          <EuiForm>
+            <EuiFormRow
+              helpText={i18n.translate('xpack.ml.datavisualizer.searchPanel.kqlEditOnlyLabel', {
+                defaultMessage: 'Currently only KQL saved searches can be edited',
+              })}
+            >
+              <EuiFieldSearch
+                value={`${searchString}`}
+                readOnly
+                data-test-subj="mlDataVisualizerLuceneSearchBarl"
+              />
+            </EuiFormRow>
+          </EuiForm>
+        )}
+      </EuiFlexItem>
+
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
+          <EuiFlexItem grow={false}>
+            <EuiSuperSelect
```

Comment:
This should have a fixed width I think. For example, it shrinks in width when selecting 'Search all', which causes the options to look like:

![image](https://user-images.githubusercontent.com/7405507/72254767-ec41fe00-35fc-11ea-8697-6c64abf07811.png)

Reply:
Fixed in https://github.com/elastic/kibana/pull/54358/commits/5be2ae4ad1f584c53632cda3966d22dd9fce506c